src/EventSubscriber/ActivityPub/UserWebFingerProfileSubscriber.php

```diff
@@ -40,28 +44,44 @@ public function buildResponse(WebfingerResponseEvent $event): void
         $params = $this->webfingerParameters->getParams();
         $jsonRd = $event->jsonRd;
 
-        if (isset($params[WebFingerParameters::ACCOUNT_KEY_NAME]) && $actor = $this->getActor(
-            $params[WebFingerParameters::ACCOUNT_KEY_NAME]
-        )) {
-            $accountHref = $this->urlGenerator->generate(
-                'ap_user',
-                ['username' => $actor->getUserIdentifier()],
-                UrlGeneratorInterface::ABSOLUTE_URL
-            );
+        if (isset($params[WebFingerParameters::ACCOUNT_KEY_NAME])) {
+            $query = $params[WebFingerParameters::ACCOUNT_KEY_NAME];
+            $this->logger->debug("got webfinger query for $query");
 
-            $link = new JsonRdLink();
-            $link->setRel('self')
-                ->setType('application/activity+json')
-                ->setHref($accountHref);
-            $jsonRd->addLink($link);
+            $domain = $this->settingsManager->get('KBIN_DOMAIN');
+            if ($domain === $query) {
+                $accountHref = $this->urlGenerator->generate('ap_instance', [], UrlGeneratorInterface::ABSOLUTE_URL);
+                $link = new JsonRdLink();
+                $link->setRel('self')
+                    ->setType('application/activity+json')
+                    ->setHref($accountHref);
+                $jsonRd->addLink($link);
+
+                return;
+            }
+
+            $actor = $this->getActor($query);
+            if ($actor) {
+                $accountHref = $this->urlGenerator->generate(
+                    'ap_user',
+                    ['username' => $actor->getUserIdentifier()],
+                    UrlGeneratorInterface::ABSOLUTE_URL
+                );
 
-            if ($actor->avatar) {
                 $link = new JsonRdLink();
-                $link->setRel('http://webfinger.net/rel/avatar')
-                    ->setHref(
-                        $this->imageManager->getUrl($actor->avatar),
-                    ); // @todo media url
+                $link->setRel('self')
+                    ->setType('application/activity+json')
+                    ->setHref($accountHref);
                 $jsonRd->addLink($link);
+
+                if ($actor->avatar) {
+                    $link = new JsonRdLink();
+                    $link->setRel('http://webfinger.net/rel/avatar')
```

Comment:
not quite sure what this hardcoded link is... oh it was in original code? I just noticed it's `http`.... but not sure how it's used

Reply:
yeah https wouldn't hurt I think